sycl/test-e2e/InvokeSimd/Regression/single_SPMD_invoke_several_nested_ESIMD.cpp

```diff
@@ -0,0 +1,140 @@
+#include <sycl/ext/oneapi/experimental/uniform.hpp>
+#include <sycl/sycl.hpp>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+
+/* Subgroup size attribute is optional.
+ * In case it is absent compiler decides what subgroup size to use.
+ */
+#ifdef IMPL_SUBGROUP
+#define SUBGROUP_ATTR
+#else
+#define SUBGROUP_ATTR [[intel::reqd_sub_group_size(VL)]]
+#endif
+
+using namespace sycl::ext::oneapi::experimental;
+namespace esimd = sycl::ext::intel::esimd;
+constexpr int VL = 16;
+
+/* *** */
+
+esimd::simd<float, VL>
+ESIMD_CALLEE_DEC(esimd::simd<float, VL> a) SYCL_ESIMD_FUNCTION {
+  return a - 1;
+}
+
+esimd::simd<float, VL>
+ESIMD_CALLEE_INC(esimd::simd<float, VL> a) SYCL_ESIMD_FUNCTION {
+  return a + 1;
+}
+
+[[intel::device_indirectly_callable]] simd<float, VL> __regcall SIMD_CALLEE_INC(
+    float *A, int i) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> a;
+  a.copy_from(A + i);
+  return ESIMD_CALLEE_INC(a);
+}
+
+/* *** */
+
+esimd::simd<float, VL> ESIMD_CALLEE_ATTUNITE(esimd::simd<float, VL> a,
+                                             bool flag) SYCL_ESIMD_FUNCTION {
+  return flag ? ESIMD_CALLEE_INC(a) : ESIMD_CALLEE_DEC(a);
+}
+
+[[intel::device_indirectly_callable]] simd<float, VL> __regcall SIMD_CALLEE(
+    float *A, int i) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> a = SIMD_CALLEE_INC(A, i);
+  return ESIMD_CALLEE_ATTUNITE(a, i < 0);
+}
+
+/* *** */
+
+using namespace sycl;
+
+int main() {
+  constexpr unsigned Size = 1024;
+  constexpr unsigned GroupSize = 4 * VL;
+
+  queue q;
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
+            << "\n";
+
+  auto *A = malloc_shared<float>(Size, q);
+  auto *B = malloc_shared<float>(Size, q);
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = i;
+    B[i] = -1;
+  }
+
+  try {
+    sycl::range<1> GlobalRange{Size};
+    // Number of workitems in each workgroup.
+    sycl::range<1> LocalRange{GroupSize};
+    sycl::nd_range<1> Range(GlobalRange, LocalRange);
+
+    auto e = q.submit([&](handler &cgh) {
+      cgh.parallel_for<class TestInvokeSimd>(
+          Range, [=](nd_item<1> ndi) SUBGROUP_ATTR {
+            sub_group sg = ndi.get_sub_group();
+            group<1> g = ndi.get_group();
+            uint32_t i = sg.get_group_linear_id() * VL +
+                         g.get_group_linear_id() * GroupSize;
+            uint32_t wi_id = i + sg.get_local_id();
+
+            float res = invoke_simd(sg, SIMD_CALLEE, uniform{A}, uniform{i});
+            res += invoke_simd(sg, SIMD_CALLEE_INC, uniform{A}, uniform{i});
+            B[wi_id] = res;
+          });
+    });
+    e.wait();
+  } catch (sycl::exception const &e) {
+    sycl::free(A, q);
+    sycl::free(B, q);
+
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    return e.code().value();
+  }
+
+  int err_cnt = 0;
+
+  for (unsigned i = 0; i < Size; ++i)
+    if (1 + 2 * A[i] != B[i])
```

Comment:
I don't understand why this test passes verification. Can we have a phone call please to discuss this test, please?